medcat/meta_cat.py

```diff
@@ -421,18 +473,37 @@ def prepare_document(self, doc: Doc, input_ids: List, offset_mapping: List, lowe
             start = ent.start_char
             end = ent.end_char
 
+            flag = 0
+            ctoken_idx = []
+            for ind, pair in enumerate(offset_mapping[last_ind:]):
+                if start <= pair[0] or start <= pair[1]:
+                    if end <= pair[1]:
+                        ctoken_idx.append(ind)
+                        break
+                    else:
+                        flag = 1
+                if flag == 1:
+                    if end <= pair[1] or end <= pair[0]:
+                        break
+                    else:
+                        ctoken_idx.append(ind)
+
             ind = 0
             # Start where the last ent was found, cannot be before it as we've sorted
             for ind, pair in enumerate(offset_mapping[last_ind:]):
                 if start >= pair[0] and start < pair[1]:
                     break
-            ind = last_ind + ind # If we did not start from 0 in the for loop
+
+            ind = last_ind + ind  # If we did not start from 0 in the for loop
             last_ind = ind
 
-            _start = max(0, ind - cntx_left)
+            # _start = max(0, ind - cntx_left)
+            _start = max(0, ctoken_idx[0] - cntx_left)
```

Comment:
Is this a bug fix?
Again, seems to affect everything, so change would need to be documented somewhere.

Reply:
Its the same change as above